pkg/kn/commands/describe.go

```diff
@@ -0,0 +1,198 @@
+
+}
+
+var boringDomains = map[string]bool{
+	"serving.knative.dev":   true,
+	"client.knative.dev":    true,
+	"kubectl.kubernetes.io": true,
+}
+
+// Write a map either compact in a single line (possibly truncated) or, if printDetails is set,
+// over multiple line, one line per key-value pair. The output is sorted by keys.
+func WriteMapDesc(dw printers.PrefixWriter, m map[string]string, label string, labelPrefix string, details bool) {
+	if len(m) == 0 {
+		return
+	}
+
+	var keys []string
+	for k := range m {
+		parts := strings.Split(k, "/")
+		if details || len(parts) <= 1 || !boringDomains[parts[0]] {
+			keys = append(keys, k)
+		}
+	}
+	if len(keys) == 0 {
+		return
+	}
+	sort.Strings(keys)
+
+	if details {
+		l := labelPrefix + label
+
+		for _, key := range keys {
+			dw.WriteColsLn(l, key+"="+m[key])
+			l = labelPrefix
+		}
+		return
+	}
+
+	dw.WriteColsLn(label, joinAndTruncate(keys, m))
+}
+
+func Age(t time.Time) string {
+	if t.IsZero() {
+		return ""
+	}
+	return duration.ShortHumanDuration(time.Now().Sub(t))
+}
+
+// Color the type of the conditions
+func formatConditionType(condition apis.Condition) string {
+	return string(condition.Type)
+}
+
+// Status in ASCII format
+func formatStatus(c apis.Condition) string {
+	switch c.Status {
+	case corev1.ConditionTrue:
+		return "++"
+	case corev1.ConditionFalse:
+		switch c.Severity {
+		case apis.ConditionSeverityError:
+			return "!!"
+		case apis.ConditionSeverityWarning:
+			return " W"
+		case apis.ConditionSeverityInfo:
+			return " I"
+		default:
+			return " !"
+		}
+	default:
+		return "??"
+	}
+}
+
+// Used for conditions table to do own formatting for the table,
+// as the tabbed writer doesn't work nicely with colors
+func getMaxTypeLen(conditions []apis.Condition) int {
+	max := 0
+	for _, condition := range conditions {
+		if len(condition.Type) > max {
+			max = len(condition.Type)
+		}
+	}
+	return max
+}
+
+// Sort conditions: Ready first, followed by error, then Warning, then Info
+func sortConditions(conditions []apis.Condition) []apis.Condition {
+	// Don't change the orig slice
+	ret := make([]apis.Condition, 0, len(conditions))
+	for _, c := range conditions {
+		ret = append(ret, c)
+	}
+	sort.SliceStable(ret, func(i, j int) bool {
+		ic := &ret[i]
+		jc := &ret[j]
+		// Ready first
+		if ic.Type == apis.ConditionReady {
+			return jc.Type != apis.ConditionReady
+		}
+		// Among conditions of the same Severity, sort by Type
+		if ic.Severity == jc.Severity {
+			return ic.Type < jc.Type
+		}
+		// Error < Warning < Info
+		switch ic.Severity {
+		case apis.ConditionSeverityError:
+			return true
+		case apis.ConditionSeverityWarning:
+			return jc.Severity == apis.ConditionSeverityInfo
+		case apis.ConditionSeverityInfo:
+			return false
+		default:
+			return false
+		}
+		return false
+	})
+	return ret
+}
+
+// Print out a table with conditions. Use green for 'ok', and red for 'nok' if color is enabled
```

Comment:
I love the idea of using colors but don't see where that's being applied here. Unless in the printer? I was expecting you would import `github.com/fatih/color` or similar?

Reply:
This is preserving something Roland planned for later use. I can delete the comment.